pkg/overlay/cache.go

```diff
@@ -399,7 +399,7 @@ func (cache *Cache) GetMissingPieces(ctx context.Context, pieces []*pb.RemotePie
 	return missingPieces, nil
 }
 
-func getIP(ctx context.Context, target string) (_ string, err error) {
+func getIP(ctx context.Context, target string) (ip string, err error) {
```

Comment:
can we have this be a net.IPAddr return value? that way GetNetwork doesn't need to call net.ParseIP